src/bulls_and_cows_lib/game.cpp

```diff
@@ -46,24 +59,66 @@ namespace bulls_and_cows {
                "    Display a message telling if the computer won or lost\n";
     }
 
-    void configure_game_options(GameOptions& game_options)
-    {
-        std::cout << "TODO:\n"
-                     "    DO\n"
-                     "       Display the current game options\n"
-                     "       Display the game options menu\n"
-                     "       Ask the user to type its choice\n"
-                     "       Treat the user's choice\n"
-                     "    UNTIL user's choice is to go back to main menu\n";
-    }
+	void configure_game_options(GameOptions& game_options)
+	{
+		bool exit = false;
+		while (!exit)
+		{
+			std::cout << "\n";
+			std::cout << "#################################" << "\n";
+			std::cout << "#################################" << "\n";
+			std::cout << "\n";
+			display_game_options(std::cout, game_options);
+			display_game_options_menu(std::cout);
+			const auto user_choice = ask_game_options_menu_choice(std::cin);
+			std::ifstream in{ "game_options.txt" };
+
+			switch (user_choice)
+			{
+			case GameOptionsMenuChoice::BackToMain:
+				exit = true;
+				break;
+			case GameOptionsMenuChoice::ModifyMaximumAllowedCharacter:
+				std::cout << "Enter the maximum character you want:";
+				game_options.maximum_allowed_character =
+					ask_char_or_default(std::cin, game_options.maximum_allowed_character);
+				break;
+			case GameOptionsMenuChoice::ModifyMaximumNumberOfAttempts:
+				std::cout << "Enter the number of attempts you want:";
+				game_options.max_number_of_attempts =
+					ask_uint_or_default(std::cin, game_options.max_number_of_attempts);
+				break;
+			case GameOptionsMenuChoice::ModifyMinimumAllowedCharacter:
+				std::cout << "Enter the minimum character you want:";
+				game_options.minimum_allowed_character =
+					ask_char_or_default(std::cin, game_options.minimum_allowed_character);
+				break;
+			case GameOptionsMenuChoice::ModifyNumberOfCharactersPerCode:
+				std::cout << "Enter the number of character per code:";
+				game_options.number_of_characters_per_code =
+					ask_uint_or_default(std::cin, game_options.number_of_characters_per_code);
+				break;
+			case GameOptionsMenuChoice::LoadOptions:
```

Comment:
pourquoi la variable in est déclarée en haut, tandis que la variable out est déclarée dans le case.
Ce serait mieux que les deux soit déclarées dans leur case respectif (on ne doit pas ouvrir le fichier tant que l'utilisateur n'a pas choisi l'option "save options").
Par contre, généralement, dès que tu déclares une variable dans un case, il faut mettre des accolades au début et à la fin du case, car la variable doit avoir un block, sinon c'est comme si la variable etait déclarée en dehors du switch.

Reply:
Done